From 3daf1e39208b73a94585b934a2a0545f0b1a2ded Mon Sep 17 00:00:00 2001
From: Vighnesh Maheshwari <vighmah@amazon.com>
Date: Mon, 1 Dec 2025 16:40:02 -0800
Subject: [PATCH] Disable loadable builtins

Signed-off-by: Vighnesh Maheshwari <vighmah@amazon.com>
---
 Makefile.in | 1 -
 1 file changed, 1 deletion(-)

diff --git a/Makefile.in b/Makefile.in
index c4a882f7..1acb647f 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -923,7 +923,6 @@ install:	.made installdirs
 		infodir=$(infodir) htmldir=$(htmldir) DESTDIR=$(DESTDIR) $@ )
 	-( cd $(DEFDIR) ; $(MAKE) $(BASH_MAKEFLAGS) DESTDIR=$(DESTDIR) $@ )
 	-( cd $(PO_DIR) ; $(MAKE) $(BASH_MAKEFLAGS) DESTDIR=$(DESTDIR) $@ )
-	-( cd $(LOADABLES_DIR) && $(MAKE) $(BASH_MAKEFLAGS) DESTDIR=$(DESTDIR) $@ )

 install-strip:
 	$(MAKE) $(BASH_MAKEFLAGS) INSTALL_PROGRAM='$(INSTALL_STRIP_PROGRAM)' \
